Flick/Controllers/EditProfileViewController.swift

```diff
@@ -319,14 +341,36 @@ class EditProfileViewController: UIViewController {
     }
 
     @objc func saveProfileInformation() {
+        // If username changed, check if username exists before updating user information
+        if let username = userNameTextField.text, username != user.username {
+            NetworkManager.checkUsernameExists(username: username) { [weak self] success in
+                guard let self = self else { return }
+                if success {
+                    self.updateUserInfo()
+                } else {
+                    self.presentInfoAlert(message: "Username invalid or already taken", completion: nil)
+                }
+            }
+        } else {
+            self.updateUserInfo()
+        }
+    }
+
+    private func updateUserInfo() {
         if let firstName = firstNameTextField.text,
             let lastName = lastNameTextField.text,
             let username = userNameTextField.text,
             let bio = bioTextView.text,
-            let base64ProfileImage = profileImageView.image?.toBase64()
-        {
-            let user = User(username: username, firstName: firstName, lastName: lastName, profilePic: base64ProfileImage, bio: bio, phoneNumber: "7812289951")
-            NetworkManager.updateUserProfile(user: user) { [weak self] userProfile in
+            // Check at least one element was updated
+            firstName != user.firstName || lastName != user.lastName || username != user.username || bio != user.bio || didChangeProfilePic {
```

Comment:
checking if at least one element is updated isn't strictly necessary probably? cause if it stayed same, making our request will still just keep profile all the same?

Reply:
you're right about that